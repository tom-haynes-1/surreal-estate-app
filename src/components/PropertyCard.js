/* eslint-disable react/jsx-one-expression-per-line */
import React from 'react';
import PropTypes from 'prop-types';
import '../styles/property-card.css';
import {
  FaRegBuilding, FaBath, FaBed, FaPoundSign,
} from 'react-icons/fa';

const PropertyCard = ({
  title,
  type,
  bathrooms,
  bedrooms,
  price,
  city,
  email,
}) => (
  <div className="property-card">
    I am a property card
    <h3 className="property-card-title">{title}</h3>
    <p className="property-card__type-city">
      <FaRegBuilding /> {type} - {city}
    </p>
    <p className="property-card__bathrooms">
      <FaBath /> {bathrooms}
    </p>
    <p className="property-card__bedrooms">
      <FaBed /> {bedrooms}
    </p>
    <p className="property-card__price">
      <FaPoundSign /> {price}
    </p>
    <div>
      <a className="property-card__email-link" href={`mailto:${email}?subject=${title}`}>
        Email
      </a>
    </div>
  </div>
);

export default PropertyCard;

PropertyCard.propTypes = {
  title: PropTypes.string.isRequired,
  type: PropTypes.string.isRequired,
  bathrooms: PropTypes.number.isRequired,
  bedrooms: PropTypes.number.isRequired,
  price: PropTypes.number.isRequired,
  city: PropTypes.string.isRequired,
  email: PropTypes.string.isRequired,
};
