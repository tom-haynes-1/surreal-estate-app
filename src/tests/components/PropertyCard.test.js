import React from 'react';
import { render, screen } from '@testing-library/react';
import PropertyCard from '../../components/PropertyCard';

describe('PropertyCard component', () => {
  const validProps = {
    title: 'test title',
    type: 'test type',
    bathrooms: 1,
    bedrooms: 2,
    price: 100000,
    city: 'test city',
    email: 'test@test.com',
  };
  test('component renders correctly', () => {
    const { asFragment } = render(
      <PropertyCard
        title={validProps.title}
        type={validProps.type}
        bathrooms={validProps.bathrooms}
        bedrooms={validProps.bedrooms}
        price={validProps.price}
        city={validProps.city}
        email={validProps.email}
      />,
    );

    expect(asFragment()).toMatchSnapshot();
  });

  test('title renders correctly', () => {
    render(
      <PropertyCard
        title={validProps.title}
        type={validProps.type}
        bathrooms={validProps.bathrooms}
        bedrooms={validProps.bedrooms}
        price={validProps.price}
        city={validProps.city}
        email={validProps.email}
      />,
    );
    const titleElement = screen.getByText('test title');

    expect(titleElement).toBeInTheDocument();
  });

  test('type and city render correctly', () => {
    render(
      <PropertyCard
        title={validProps.title}
        type={validProps.type}
        bathrooms={validProps.bathrooms}
        bedrooms={validProps.bedrooms}
        price={validProps.price}
        city={validProps.city}
        email={validProps.email}
      />,
    );
    const typeElement = screen.getByText('test type - test city');

    expect(typeElement).toBeInTheDocument();
  });

  test('bathrooms renders correctly', () => {
    render(
      <PropertyCard
        title={validProps.title}
        type={validProps.type}
        bathrooms={validProps.bathrooms}
        bedrooms={validProps.bedrooms}
        price={validProps.price}
        city={validProps.city}
        email={validProps.email}
      />,
    );
    const bathroomElement = screen.getByText('1');

    expect(bathroomElement).toBeInTheDocument();
  });

  test('bedroom element renders correctly', () => {
    render(
      <PropertyCard
        title={validProps.title}
        type={validProps.type}
        bathrooms={validProps.bathrooms}
        bedrooms={validProps.bedrooms}
        price={validProps.price}
        city={validProps.city}
        email={validProps.email}
      />,
    );
    const bedroomElement = screen.getByText('2');

    expect(bedroomElement).toBeInTheDocument();
  });

  test('price renders correctly', () => {
    render(
      <PropertyCard
        title={validProps.title}
        type={validProps.type}
        bathrooms={validProps.bathrooms}
        bedrooms={validProps.bedrooms}
        price={validProps.price}
        city={validProps.city}
        email={validProps.email}
      />,
    );
    const priceElement = screen.getByText('100000');

    expect(priceElement).toBeInTheDocument();
  });

  test('email link renders correctly', () => {
    render(
      <PropertyCard
        title={validProps.title}
        type={validProps.type}
        bathrooms={validProps.bathrooms}
        bedrooms={validProps.bedrooms}
        price={validProps.price}
        city={validProps.city}
        email={validProps.email}
      />,
    );
    const emailElement = screen.getByText('Email');

    expect(emailElement).toBeInTheDocument();
  });
});
