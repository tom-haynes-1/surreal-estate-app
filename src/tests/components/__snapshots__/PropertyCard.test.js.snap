// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PropertyCard component component renders correctly 1`] = `
<DocumentFragment>
  <div
    class="property-card"
  >
    I am a property card
    <h3
      class="property-card-title"
    >
      test title
    </h3>
    <p
      class="property-card__type-city"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 448 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M128 148v-40c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v40c0 6.6-5.4 12-12 12h-40c-6.6 0-12-5.4-12-12zm140 12h40c6.6 0 12-5.4 12-12v-40c0-6.6-5.4-12-12-12h-40c-6.6 0-12 5.4-12 12v40c0 6.6 5.4 12 12 12zm-128 96h40c6.6 0 12-5.4 12-12v-40c0-6.6-5.4-12-12-12h-40c-6.6 0-12 5.4-12 12v40c0 6.6 5.4 12 12 12zm128 0h40c6.6 0 12-5.4 12-12v-40c0-6.6-5.4-12-12-12h-40c-6.6 0-12 5.4-12 12v40c0 6.6 5.4 12 12 12zm-76 84v-40c0-6.6-5.4-12-12-12h-40c-6.6 0-12 5.4-12 12v40c0 6.6 5.4 12 12 12h40c6.6 0 12-5.4 12-12zm76 12h40c6.6 0 12-5.4 12-12v-40c0-6.6-5.4-12-12-12h-40c-6.6 0-12 5.4-12 12v40c0 6.6 5.4 12 12 12zm180 124v36H0v-36c0-6.6 5.4-12 12-12h19.5V24c0-13.3 10.7-24 24-24h337c13.3 0 24 10.7 24 24v440H436c6.6 0 12 5.4 12 12zM79.5 463H192v-67c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v67h112.5V49L80 48l-.5 415z"
        />
      </svg>
       test type - test city
    </p>
    <p
      class="property-card__bathrooms"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M32,384a95.4,95.4,0,0,0,32,71.09V496a16,16,0,0,0,16,16h32a16,16,0,0,0,16-16V480H384v16a16,16,0,0,0,16,16h32a16,16,0,0,0,16-16V455.09A95.4,95.4,0,0,0,480,384V336H32ZM496,256H80V69.25a21.26,21.26,0,0,1,36.28-15l19.27,19.26c-13.13,29.88-7.61,59.11,8.62,79.73l-.17.17A16,16,0,0,0,144,176l11.31,11.31a16,16,0,0,0,22.63,0L283.31,81.94a16,16,0,0,0,0-22.63L272,48a16,16,0,0,0-22.62,0l-.17.17c-20.62-16.23-49.83-21.75-79.73-8.62L150.22,20.28A69.25,69.25,0,0,0,32,69.25V256H16A16,16,0,0,0,0,272v16a16,16,0,0,0,16,16H496a16,16,0,0,0,16-16V272A16,16,0,0,0,496,256Z"
        />
      </svg>
       1
    </p>
    <p
      class="property-card__bedrooms"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 640 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M176 256c44.11 0 80-35.89 80-80s-35.89-80-80-80-80 35.89-80 80 35.89 80 80 80zm352-128H304c-8.84 0-16 7.16-16 16v144H64V80c0-8.84-7.16-16-16-16H16C7.16 64 0 71.16 0 80v352c0 8.84 7.16 16 16 16h32c8.84 0 16-7.16 16-16v-48h512v48c0 8.84 7.16 16 16 16h32c8.84 0 16-7.16 16-16V240c0-61.86-50.14-112-112-112z"
        />
      </svg>
       2
    </p>
    <p
      class="property-card__price"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 320 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M308 352h-45.495c-6.627 0-12 5.373-12 12v50.848H128V288h84c6.627 0 12-5.373 12-12v-40c0-6.627-5.373-12-12-12h-84v-63.556c0-32.266 24.562-57.086 61.792-57.086 23.658 0 45.878 11.505 57.652 18.849 5.151 3.213 11.888 2.051 15.688-2.685l28.493-35.513c4.233-5.276 3.279-13.005-2.119-17.081C273.124 54.56 236.576 32 187.931 32 106.026 32 48 84.742 48 157.961V224H20c-6.627 0-12 5.373-12 12v40c0 6.627 5.373 12 12 12h28v128H12c-6.627 0-12 5.373-12 12v40c0 6.627 5.373 12 12 12h296c6.627 0 12-5.373 12-12V364c0-6.627-5.373-12-12-12z"
        />
      </svg>
       100000
    </p>
    <div>
      <a
        class="property-card__email-link"
        href="mailto:test@test.com?subject=test title"
      >
        Email
      </a>
    </div>
  </div>
</DocumentFragment>
`;
